core/controller/src/main/scala/whisk/core/controller/AuthorizedRouteDispatcher.scala

```diff
@@ -71,7 +74,21 @@ trait BasicAuthorizedRouteProvider extends Directives {
 
     onComplete(entitlementProvider.check(user, right, resource)) {
       case Success(_) => dispatchOp(user, right, resource)
-      case Failure(t) => handleEntitlementFailure(t)
+      case Failure(t) =>
+        t match {
+          case (r: RejectRequest) =>
+            r.code match {
+              case Forbidden =>
+                handleEntitlementFailure(
+                  RejectRequest(
+                    Forbidden,
+                    Some(ErrorResponse(Messages.notAuthorizedtoAccessResource(resource.fqname), transid))))
+              case NotFound =>
+                handleEntitlementFailure(
+                  RejectRequest(NotFound, Some(ErrorResponse(Messages.resourceDoesntExist(resource.fqname), transid))))
+              case _ => handleEntitlementFailure(t)
+            }
+        }
```

Comment:
Does this disambiguate access to your own namespace vs. access to any other namespace? We shall not return `not found` if we don't even have access to that namespace because then we'd leak information from other peoples namespaces.

Reply:
@markusthoemmes - not sure i completely understand the question.  if a user does not have authority to access the targeted `namespace + entity` or `namespace + package + entity`, a forbidden response will be returned - even if the namespace does not exist.  from what i've seen, the target resource name displayed will be dependent on the resource being checked.  that value might be `ns/pkg` or `ns` or `ns/entity`.

Reply:
@markusthoemmes the entitlement should yield different rejection codes (forbidden or not found) depending on the subject and their rights. There should be tests for this.

Reply:
_thinking out loud_  to limit the amount of information giving to an attacker, handing out a single response for either forbidden or not found seems like a good way to avoid resource name discovery.

Reply:
In general yes but here the resources are explicitly named in the request. I’m not sure I see the point of hiding details beyond not found vs forbidden.